lib/jsonapi/authorization/default_pundit_authorizer.rb

```diff
@@ -191,9 +238,9 @@ def remove_to_many_relationship(source_record, related_record, relationship_type
       #
       # * +source_record+ - The record whose relationship is modified
       # * +relationship_type+ - The relationship type
-      def remove_to_one_relationship(source_record, relationship_type)
+      def remove_to_one_relationship(source_record, related_record, relationship_type)
         relationship_method = "remove_#{relationship_type}?"
-        authorize_relationship_operation(source_record, relationship_method)
+        authorize_relationship_operation(source_record, relationship_method, related_record)
```

Comment:
Removing an association doesn't need the old records, as it will be available directly from the record inside the Pundit policies. Wasn't that what you discussed with @matthias-g in https://github.com/venuu/jsonapi-authorization/pull/40#issuecomment-280493499?